crates/bevy_ecs/src/world/mod.rs

```diff
@@ -3116,19 +3109,27 @@ impl World {
     /// Runs both [`clear_entities`](Self::clear_entities) and [`clear_resources`](Self::clear_resources),
     /// invalidating all [`Entity`] and resource fetches such as [`Res`](crate::system::Res), [`ResMut`](crate::system::ResMut)
     pub fn clear_all(&mut self) {
-        self.clear_entities();
-        self.clear_resources();
-    }
-
-    /// Despawns all entities in this [`World`].
-    pub fn clear_entities(&mut self) {
         self.storages.tables.clear();
         self.storages.sparse_sets.clear_entities();
         self.archetypes.clear_entities();
         self.entities.clear();
         self.allocator.restart();
     }
 
+    /// Despawns all entities in this [`World`].
+    pub fn clear_entities(&mut self) {
```

Comment:
This function should clear all entities (including resources), as resources _are_ entities. Filtering out specific entities to clear (ex: resources) should be done via a separate API.